pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -301,7 +301,7 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 		volumeName = util.GetUniqueVolumeNameFromSpecWithPod(podName, volumePlugin, volumeSpec)
 	}
 
-	seLinuxFileLabel, pluginSupportsSELinuxContextMount, err := dsw.getSELinuxLabel(volumeSpec, seLinuxContainerContexts)
+	seLinuxFileLabel, pluginSupportsSELinuxContextMount, err := dsw.getSELinuxLabel(volumeSpec, seLinuxContainerContexts, pod.Spec.SecurityContext)
```

Comment:
Not visible in this PR, but should `GetPodVolumeNames` be changed to return selinux context for newer featuregate as well? Currently I think it will only return `seLinuxContainerContexts` for `ReadWriteOnce` feature-gate enabled.

Reply:
`GetPodVolumeNames` returns all pod volumes and all their labels, regardless of `SELinuxMount` feature gate.

Caller of `GetPodVolumeNames` must check the gates to produce the right errors / warnings / metrics if the access mode + feature gate do not match.

Reply:
Ok, I moved it out. Most `GetPodVolumeNames` don't need to collect SELinuxOptions and those who do, they can evaluate the feature gate before the call.